spec/policies/user_policy_spec.rb

```diff
@@ -86,6 +92,72 @@
     end
   end
 
+  context "for super org admins" do
+    permissions :index? do
+      it "is allowed for super org admins" do
+        expect(subject).to permit(build(:super_org_admin), User)
+      end
+    end
+
+    primary_management_actions.each do |permission|
+      permissions permission do
+        it "is forbidden for super org admins" do
+          expect(subject).not_to permit(build(:super_org_admin), User)
+        end
+      end
+    end
+
+    disallowed_actions_super_org_admin.each do |disallowed_super_org_admin_permission|
+      permissions disallowed_super_org_admin_permission do
+        it "is forbidden for super org admins to create any type of user or assign orgs to them" do
+          expect(subject).not_to permit(super_org_admin, build(:superadmin_user))
+          expect(subject).not_to permit(super_org_admin, build(:admin_user))
+          expect(subject).not_to permit(super_org_admin, build(:super_org_admin))
+          expect(subject).not_to permit(super_org_admin, build(:organisation_admin))
+          expect(subject).not_to permit(super_org_admin, build(:user_in_organisation))
+          expect(subject).not_to permit(super_org_admin, build(:user_in_organisation, organisation: organisation_admin.organisation))
+        end
+      end
+    end
+
+    super_org_admin_actions.each do |allowed_super_org_admin_permission|
+      permissions allowed_super_org_admin_permission do
+        it "is allowed for super org admins to access normal users from within their own organisation" do
+          expect(subject).not_to permit(super_org_admin, build(:user_in_organisation))
+        end
+
+        it "is forbidden for super org admins to access normal users from other organisations" do
+          expect(subject).to permit(super_org_admin, build(:user_in_organisation, organisation: super_org_admin.organisation))
+        end
+
+        it "is allowed for super org admins to access org admins from within their own organisation" do
+          expect(subject).to permit(super_org_admin, build(:organisation_admin, organisation: super_org_admin.organisation))
+        end
+
+        it "is allowed for super org admins to access super org admins from within their own organisation" do
+          expect(subject).to permit(super_org_admin, build(:super_org_admin, organisation: super_org_admin.organisation))
+        end
+
+        it "is forbidden for super org admins to access org admins from other organisations" do
+          expect(subject).to permit(super_org_admin, build(:organisation_admin, organisation: super_org_admin.organisation))
+        end
+
+        it "is forbidden for super org admins to access superadmins and admin users from either their own or other organisations" do
+          expect(subject).not_to permit(super_org_admin, build(:superadmin_user))
+          expect(subject).not_to permit(super_org_admin, build(:admin_user))
+        end
+      end
+    end
```

Comment:
I think we're missing a test that shows a super org admin is allowed to do things to users within their org's subtree.

Reply:
And maybe a corresponding test in the org admin tests that make sure they can't do things to users within their org's subtree.  It has "forbidden to access other orgs" but we could do "forbidden to access child orgs" to be explicit.